tlatools/org.lamport.tlatools/test/pcal/parser/PcalParserTranslator.java

```diff
@@ -0,0 +1,184 @@
+import pcal.PcalCharReader;
+import pcal.TLAtoPCalMapping;
+import pcal.TLAExpr;
+import pcal.TLAToken;
+import pcal.ParseAlgorithm;
+import pcal.PcalParams;
+import pcal.exception.ParseAlgorithmException;
+
+import org.junit.Assert;
+
+/**
+ * Utilities for parsing PlusCal and converting it to a form compatible with
+ * the unified TLA+ syntax parser test corpus.
+ */
+public class PcalParserTranslator {
+	
+	/**
+	 * Simple data class holding info about the start of a PlusCal block.
+	 */
+	private static class PcalStart {
+		/**
+		 * The line on which the PlusCal block starts.
+		 */
+		int line = 0;
+		
+		/**
+		 * The column in which the PlusCal block starts, directly after the
+		 * --algorithm or --fair algorithm tokens.
+		 */
+		int column = 0;
+		
+		/**
+		 * Whether the PlusCal block is a fair algorithm.
+		 */
+		boolean isFair = false;
+		
+		/**
+		 * Finds the start of the PlusCal block.
+		 * 
+		 * @param input Text containing PlusCal block, split into lines.
+		 */
+		public PcalStart(Vector<String> input) {
+			final String fairStart = "--fair";
+			final String unfairStart = "--algorithm";
+			for (int i = 0; i < input.size(); i++) {
+				String line = input.get(i);
+				int keywordStart = line.indexOf(fairStart);
+				if (-1 == keywordStart) {
+					keywordStart = line.indexOf(unfairStart);
+					if (-1 == keywordStart) {
+						continue;
+					} else {
+						this.line = i;
+						this.column = keywordStart + unfairStart.length();
+						this.isFair = false;
+						break;
+					}
+				} else {
+					this.line = i;
+					this.column = keywordStart + fairStart.length();
+					this.isFair = true;
+					break;
+				}
+			}
+		}
+	}
+	
+	/**
+	 * Calls the PlusCal parser with the given input string.
+	 * 
+	 * @param input The unparsed PlusCal text.
+	 * @return A parsed PlusCal Abstract Syntax Tree (AST).
+	 * @throws ParseAlgorithmException If the PlusCal syntax is invalid.
+	 */
+	public static AST parse(String input) throws ParseAlgorithmException {
+		Vector<String> lines = new Vector<String>(Arrays.asList(input.split("\r?\n")));
+		PcalStart start = new PcalStart(lines);
+
+		// Some static PlusCal parser initialization
+		TLAtoPCalMapping mapping = new TLAtoPCalMapping() ;
+		mapping.algLine = start.line;
+		mapping.algColumn = start.column;
+		PcalParams.tlaPcalMapping = mapping;
+
+		// Parse algorithm
+		int endLine = lines.size();
+		int endColumn = (int)lines.lastElement().codePoints().count();
+		PcalCharReader reader = new PcalCharReader(lines, start.line, start.column, endLine, endColumn);
+		return ParseAlgorithm.getAlgorithm(reader, start.isFair);
+	}
+	
+	/**
+	 * Translates the given TLA+ token. We just hardcode a small set of
+	 * tokens used as expression placeholders in the test corpus.
+	 * 
+	 * @param token The TLA+ token to translate.
+	 * @return The corresponding node in normalized form.
+	 */
+	public static AstNode translate(TLAToken token) {
+		switch (token.string) {
+			case "TRUE": {
```

Comment:
The docstring should list which tokens are in the "small set" this method can translate.